app/lib/actions/katello/content_view/publish.rb

```diff
@@ -181,11 +184,15 @@ def version_for_publish(content_view, options)
         end
 
         def handle_import(version, path:, metadata:)
-          plan_action(::Actions::Pulp3::Orchestration::ContentViewVersion::Import, version, path: path, metadata: metadata)
-          plan_action(::Actions::Pulp3::Orchestration::ContentViewVersion::CopyVersionUnitsToLibrary, version)
-          concurrence do
-            version.importable_repositories.pluck(:id).each do |id|
-              plan_action(Katello::Repository::IndexContent, id: id)
+          return if @imported
+          @imported = true
```

Comment:
Why was this needed?

I heard once that I shouldn't use class instance variables in dynflow actions, because you can't be 100% sure that it's always the same instance.  Instead it's more reliable to use `input` or `output`.  (Could probably do a `plan_self` at the end of `plan` and then check here for `input[:imported]`, something like that.

Reply:
```rb
handle_import(version, options.slice(:path, :metadata))
```
Looks like this doesn't even depend on `repositories`.  Can it be moved outside the `each` loop?

If not, I still think `input[:imported]` instead of `@imported` would work fine.  Once you've set an input within `plan`, you can query it as long as it doesn't depend on any other step's output.

As an alternative you could set a local variable (`imported = false`) outside the `each` loop and then set `imported = true` after.  Like you have above except the initialization is outside the loop.  That doesn't seem bad to me.

Reply:
Updated :)